scapy/packet.py

```diff
@@ -838,9 +847,12 @@ def guess_payload_class(self, payload):
         """
         for t in self.aliastypes:
             for fval, cls in t.payload_guess:
-                if all(hasattr(self, k) and v == self.getfieldval(k)
-                       for k, v in six.iteritems(fval)):
-                    return cls
+                try:
+                    if all(v == self.getfieldval(k)
+                           for k, v in six.iteritems(fval)):
+                        return cls
+                except AttributeError:
+                    pass
```

Comment:
This is actually quite a big improvement: in fact `hasattr` calls `__getattr__` under the hood, which calls `getfieldval` => we actually did `getfieldval` twice each time.